### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by valentinewallace**
Matt recently left a message on Slack going into LDK's customizability. It was decided in the LDK biweekly dev meeting that this message should be added in the README. I also added a few other updates to the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
 
+The `lightning-persister` crate implements persistence for channel data that
+is crucial to avoiding loss of channel funds. Sample modules for persistence of
+other Rust-Lightning data is coming soon.
+
 Status
 ------
 
 The project implements all of the BOLT specifications in the 1.0 spec except
 for [channel queries](https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#query-messages). The
```
Comment (TheBlueMatt) [inline README.md:25]: Oh, I didnt realize this was here - this is out of date now.

File: README.md [hunk 2]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+LDK is likely your only option.
+
+We are currently working on a demo node which fetches blockchain data and
+on-chain funds via bitcoin core RPC/REST. The individual pieces of that demo
+are/will be composable, so you can pick the off-the-shelf parts you want and
+replace the rest.
+
+In general, Rust-Lightning does not provide (but LDK has implementations of):
+* on-disk storage - you can store the channel state any way you want - whether
+  google drive/icloud, a local disk, any K-V store/database/a remote server, or
```
Comment (TheBlueMatt) [inline README.md:56]: nit: Google Drive/iCloud.

File: README.md [hunk 3]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+  any combination of them - we provide a clean api that provides objects which
```
Comment (TheBlueMatt) [inline README.md:57]: nit API

File: README.md [hunk 4]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+* blockchain data - we provide a simple `block_connected`/`block_disconnected`
+  API which you provide block headers and transaction information to. We also
+  provide an API for getting information about transactions we wish to be
+  informed of, which is compatible with Electrum server requests/neutrino
+  filtering/etc.
+* UTXO management - RL/LDK owns on-chain funds as long as they are claimable as
+  a part of a lightning output which can be contested - once a channel is closed
+  and all on-chain outputs are spendable only by the user, we provide users
+  notifications that a UTXO is "theirs" again and it is up to them to spend it
+  as they wish. Additionally, we do not provide a way to fund channels aside
```
Comment (TheBlueMatt) [inline README.md:68]: s/we do not provide a way to fund channels aside from a generic API/channel funding is accomplished with a generic API/

File: README.md [hunk 5]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+  from a generic API which notifies users of exactly what output they need to
+  create a transaction for, and we handle the rest. This is a large part of our
```
Comment (TheBlueMatt) [inline README.md:70]: s/exactly what output they need to create a transaction for/the output which needs to appear on-chain, and once a transaction is created/

File: README.md [hunk 6]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
 dependencies with dependencies. Do convince Andrew to cut down dependency usage
 in rust-bitcoin.
 
+Rust-Lightning vs. LDK (Lightning Development Kit)
+-------------
+Rust-Lightning refers to the core `lightning` crate within this repo, whereas
+LDK encompasses Rust-Lightning and all of its sample modules and crates (e.g.
+the `lightning-persister` crate), language bindings, sample node
+implementation(s), and other tools built around using Rust-Lightning for
+lightning integration or building a lightning node.
```
Comment (TheBlueMatt) [inline README.md:110]: Maybe mention language bindings.

**Review by TheBlueMatt — APPROVED** [review]
A few nits but it looks good, thanks for doing this! cc @moneyball.

File: README.md [hunk 7]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+production-worthy implementation in 2021.
 
 Communications for Rust-Lightning and Lightning Development Kit happens through
 [LDK slack](http://lightningdevkit.org/).
 
+About
+-----------
+LDK/Rust-Lightning is a generic library which allows you to build a lightning
+node without needing to worry about getting all of the lightning state machine
+and on-chain punishment code exactly correct. Note that Rust-Lightning isn't, in
```
Comment (ariard) [inline README.md:41]: I would switch to "on-chain reaction", really on-chain stuff is far wider than punishment. Also precise routing and HTLC forwarding. E.g Electrum doesn't implement forwarding IIRC, so a good point to mention.

File: README.md [hunk 8]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+itself, a node. There are various working/in progress demos which could be used
+as a node today, but if you "just" want a generic lightning node, you're almost
+certainly better off with `c-lightning`/`lnd` - if, on the other hand, you want
+to integrate lightning with custom features such as your own chain sync, your
+own key mangement, your own hardware, your own data storage/backup logic, etc.,
```
Comment (ariard) [inline README.md:46]: We should mention leveraging library utils like BOLT8/`chan_utils`. At least I some point I'm expecting some _non-LN_ Bitcoin applications reusing our P2P stack (e.g DLC, bulletin boards, ...).

File: README.md [hunk 9]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+  a part of a lightning output which can be contested - once a channel is closed
+  and all on-chain outputs are spendable only by the user, we provide users
+  notifications that a UTXO is "theirs" again and it is up to them to spend it
+  as they wish. Additionally, we do not provide a way to fund channels aside
+  from a generic API which notifies users of exactly what output they need to
+  create a transaction for, and we handle the rest. This is a large part of our
+  API's goals - making it easier to integrate lightning into existing on-chain
+  wallets which have their own on-chain logic - without needing to move funds in
+  and out of a "lightning subwallet" with on-chain transactions and a separate
+  private key system.
```
Comment (ariard) [inline README.md:74]: "However, as part of the anchor output support, the library will require to lockdown "reserves UTXO" to cover emergency onchain fees".

File: README.md [hunk 10]
```diff
@@ -8,47 +8,106 @@ Rust-Lightning
...
+in mind, everything should be exposed via simple, composable APIs. More
+information about Rust-Lightning's flexibility is provided in the `About`
+section above.
 
 For security reasons, do not add new dependencies. Really do not add new
 non-optional/non-test/non-library dependencies. Really really do not add
 dependencies with dependencies. Do convince Andrew to cut down dependency usage
 in rust-bitcoin.
 
+Rust-Lightning vs. LDK (Lightning Development Kit)
```
Comment (ariard) [inline README.md:104]: Finally! The most embarrassing and recurring question I've been asked the past year :smile:

**Review by ariard — COMMENTED** [review]
Great addition!

File: README.md (hunk 6 above)
Comment (valentinewallace) [reply README.md:110]: They're in there^ should it be rephrased?

File: README.md (hunk 10 above)
Comment (valentinewallace) [reply README.md:104]: I know right. We should eventually have an FAQ section probably

File: README.md (hunk 9 above)
Comment (valentinewallace) [reply README.md:74]: That seems like it requires a good bit more explanation and docs. Could we push this to a follow-up to more thoroughly explain it?

File: README.md (hunk 7 above)
Comment (valentinewallace) [reply README.md]: Isn't forwarding implied when we say we're fully BOLT-compliant? Added routing though and amended the "on-chain" language.

File: README.md (hunk 6 above)
Comment (TheBlueMatt) [reply README.md:112]: Its ok, I'm just blind.

File: README.md (hunk 7 above)
Comment (TheBlueMatt) [reply README.md]: I don't think "BOLT-compliant" is well-defined - I'd call a client that doesn't do forwarding compliant.

Comment (ariard) [reply README.md]: Agree, forwarding/routing are what make Lightning actually a network, without it's just nice bidirectional payment channel.

File: README.md (hunk 9 above)
Comment (ariard) [reply README.md]: Let's push it to a follow-up, anchor output will need some good discussion around interfacing with L1 keys/utxos.

**Comment by ariard** [comment]
ACK 2535695, thanks for updating this.

**Pull request merged**